cirq/ops/gate_operation.py

```diff
@@ -85,7 +85,15 @@ def with_gate(self, new_gate: raw_types.Gate) -> 'GateOperation':
         return new_gate.on(*self.qubits)
 
     def __repr__(self):
-        return 'GateOperation({!r}, {!r})'.format(self.gate, self.qubits)
+        # Abbreviate when possible.
+        if self == self.gate.on(*self.qubits):
+            return '{!r}.on({})'.format(
+                self.gate,
+                ', '.join(repr(q) for q in self.qubits))
```

Comment:
This can generate a repr that can't be evaulated.  Maybe use parenthesis if it's not a just letters.
```
>>> repr(cirq.X(q0) ** 0.5)
X**0.5.on(NamedQubit('q0'))
```

Reply:
I think this should be a separate issue on `EigenGate` not using parens in its repr.

Reply:
All the repr changes probably shouldn't be part of this PR anyways.

Reply:
Done.